contrib/docker/Dockerfile.apache

```diff
@@ -33,74 +32,65 @@ RUN apt-get install -y --no-install-recommends \
 ## Image Processing
       libjpeg62-turbo-dev \
       libpng-dev \
+      libmagickwand-dev \
 # Required for GD
       libxpm4 \
       libxpm-dev \
       libwebp6 \
       libwebp-dev \
 ## Video Processing
-      ffmpeg
-
-# Update Local data
-RUN sed -i '/en_US/s/^#//g' /etc/locale.gen && locale-gen && update-locale
-
+      ffmpeg \
+## Database
+#      libpq-dev \
+#      libsqlite3-dev \
+# Locales Update
+  && sed -i '/en_US/s/^#//g' /etc/locale.gen \
+  && locale-gen \
+  && update-locale \
 # Install PHP extensions
-RUN docker-php-source extract
-
+  && docker-php-source extract \
 #PHP Imagemagick extensions
-RUN apt-get install -y --no-install-recommends libmagickwand-dev
-RUN pecl install imagick
-RUN docker-php-ext-enable imagick
-
+  && pecl install imagick \
+  && docker-php-ext-enable imagick \
 # PHP GD extensions
-RUN docker-php-ext-configure gd \
+  && docker-php-ext-configure gd \
       --with-freetype \
       --with-jpeg \
       --with-webp \
-      --with-xpm
-RUN docker-php-ext-install -j$(nproc) gd
-
+      --with-xpm \
+  && docker-php-ext-install -j$(nproc) gd \
 #PHP Redis extensions
-RUN pecl install redis
-RUN docker-php-ext-enable redis
-
+  && pecl install redis \
+  && docker-php-ext-enable redis \
 #PHP Database extensions
-RUN apt-get install -y --no-install-recommends libpq-dev libsqlite3-dev
-RUN docker-php-ext-install pdo_mysql pdo_pgsql pdo_sqlite
-
+  && docker-php-ext-install pdo_mysql \
+#pdo_pgsql pdo_sqlite \
 #PHP extensions (dependencies)
-RUN docker-php-ext-configure intl
-RUN docker-php-ext-install -j$(nproc) intl bcmath zip pcntl exif curl
-
+  && docker-php-ext-configure intl \
+  && docker-php-ext-install -j$(nproc) intl bcmath zip pcntl exif curl \
 #APACHE Bootstrap
-RUN a2enmod rewrite remoteip \
+  && a2enmod rewrite remoteip \
  && {\
      echo RemoteIPHeader X-Real-IP ;\
      echo RemoteIPTrustedProxy 10.0.0.0/8 ;\
      echo RemoteIPTrustedProxy 172.16.0.0/12 ;\
      echo RemoteIPTrustedProxy 192.168.0.0/16 ;\
      echo SetEnvIf X-Forwarded-Proto "https" HTTPS=on ;\
     } > /etc/apache2/conf-available/remoteip.conf \
- && a2enconf remoteip
-
+ && a2enconf remoteip \
 #Cleanup
-RUN docker-php-source delete
-RUN apt-get autoremove --purge -y
-RUN apt-get clean
-RUN rm -rf /var/cache/apt
-RUN rm -rf /var/lib/apt/lists/*
-
-ENV PATH="~/.composer/vendor/bin:./vendor/bin:${PATH}"
+  && docker-php-source delete \
+  && apt-get autoremove --purge -y \
+  && apt-get clean \
+  && rm -rf /var/cache/apt \
+  && rm -rf /var/lib/apt/lists/
 
 COPY . /var/www/
-WORKDIR /var/www/
-
-RUN cp -r storage storage.skel
-RUN composer global require hirak/prestissimo --no-interaction --no-suggest --prefer-dist
-RUN composer install --prefer-dist --no-interaction --no-ansi --optimize-autoloader
-RUN composer global remove hirak/prestissimo
-RUN rm -rf html && ln -s public html
-
+RUN cp -r storage storage.skel \
```

Comment:
why is the actual storage copied to the skeleton dir. is that a bug?

Reply:
Yeah, I find this weird as well...
I'm quite confused, likely just going to recreate a proper Docker build for my fork

Reply:
I guess it's to pull in future updates to the shipped storage to your local storage volume. 
`storage` here is the one from the git checkout, where `storage` on your running container is your volume. 
By copying `storage.skel` back to `storage` in the entrypoint, it automatically updates the local storage in case the image ships with a new subdirectory or something.

Reply:
but `storage` is the source and its content is copied to the target `storage.skel`

Reply:
Nobody knows why this exists but everybody is too scared to change it ;)

leave it as is if we want to get this merged and we can test later.

Reply:
Yes, it's the source from github. That's the part we want to keep to update our volumes from past version.

Let me try to construct an example:
- We ship `storage/` populated with some subdirectories: https://github.com/pixelfed/pixelfed/tree/dev/storage
- Next step, you download the repo, build your image, spin up your container and mount storage as a volume, because you want to keep your uploads.
- On Container+Volume creation, the content of `storage/ (github)` get copied to `storage/ (your volume)`, but only on the first time
- Now, on a future update, `storage/ (github)` introduces a new subdirectory
- When you update your existing deployment, and mount your old `storage/ (volume)` back, you would be missing that new subdirectory

That's why we always copy `storage/ (github)` to that skel directory on image creation, and back on container creation (in the entrypoint). So you'll always get all of github-storages content in your volume.

If you want we can continue this on IRC, I'm hansenerd, that'd be likely a lot faster ;)

Reply:
A real world example that I just noticed: If cities.json would be updated in this repo, it would never reach your existing volume if it weren't for the skel workaround.